### PR TITLE
feat: add credential management

### PR DESCRIPTION
**Pull request by g-linville**

### CODE REVIEW

File: src/gptscript.ts [hunk 1]
```diff
@@ -320,6 +320,70 @@ export class GPTScript {
...
     }
 
+    async listCredentials(context: Array<string>, allContexts: boolean): Promise<Array<Credential>> {
+        if (!this.ready) {
+            this.ready = await this.testGPTScriptURL(20)
+        }
+        const resp = await fetch(`${GPTScript.serverURL}/credentials`, {
+            method: "POST",
+            body: JSON.stringify({context, allContexts})
+        })
```
Comment (thedadams) [inline src/gptscript.ts]: Does `runSubcommand` not work for these?

Comment (g-linville) [reply src/gptscript.ts]: I remember trying it and it not working for some reason, but I don't remember why...let me try that again and see.

Comment (g-linville) [reply src/gptscript.ts]: Taking a look at the code again, `RunSubcommand` doesn't let me specify an arbitrary request body to use. I can only pass it a tool and some run options. But I need to be able to pass things like a credential (in the case of create), or name/context information for the other operations.

Comment (g-linville) [reply src/gptscript.ts]: Let me know if you want me to change this or leave it as is @thedadams

Comment (thedadams) [reply src/gptscript.ts]: Can you try something like this?
```
const r: Run = new RunSubcommand("credentials", undefined, {}, GPTScript.serverURL)

r.request(payload)
await r.text()
```

Comment (g-linville) [reply src/gptscript.ts]: That worked, thanks. This is updated now

**Review by thedadams — COMMENTED** [review]
Can you also add the `CredentialContexts` stuff here, too?

**Comment by g-linville** [comment]
> Can you also add the `CredentialContexts` stuff here, too?

Done

**Review by StrongMonkey — APPROVED** [review]

**Review by thedadams — APPROVED** [review]

**Pull request merged**